board: xhsc: hc32f4a0_evb: Add minimal supprot for hc32f4a0_evb board

XH-YongbingLiu opened the pull request:
This commit adds support for the hc32f4a0_evb board to print info by usart, including the necessary devicetree files, Kconfig configurations, peripherals and board-specific documentation.

The following changes about the board are included:

    Add hc32f4a0_evb.dts with hardware description for the board.
    Add Kconfig.board and Kconfig.defconfig for default software configurations.
    Add board documentation in doc/index.rst. - Add board.yaml for test runner metadata.

To implement console output, the following peripheral are supported:

    gpio
    intc
    clock
    serial

nordicjm commented on drivers/clock_control/Kconfig.hc32:
```diff
@@ -0,0 +1,34 @@
+	  Select medium Driver for XTAL32
+endchoice
+
+config HC32_XTAL32_DRV
+	int
+	default 0 if HC32_XTAL32_DRV_M
+	default 1 if HC32_XTAL32_DRV_H
```
make them dts properties

XH-YongbingLiu committed:
manifest: Add hal_xhsc to HAL group.

Add HAL repository information for the first supported HC32F4A0_EVB
and HC32F460_EVB.

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

XH-YongbingLiu committed:
drivers: clock_control: Add support for HC32F4A0

This is the initial commit to support clock_control driver for
XHSC HC32F4A0

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

XH-YongbingLiu committed:
drivers: gpio: Add support for HC32F4A0

This is the initial commit to support gpio driver for XHSC HC32F4A0

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

XH-YongbingLiu committed:
drivers: pinctrl: Add support for HC32F4A0

This is the initial commit to support pinctrl driver for XHSC HC32F4A0

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

XH-YongbingLiu committed:
drivers: interrupt_controller: Add support for HC32F4A0

This is the initial commit to support interrupt controller

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

XH-YongbingLiu committed:
drivers: serial: Add support for HC32F4A0

This is the initial commit to support USART driver for XHSC HC32F4A0

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

XH-YongbingLiu committed:
soc: xhsc: Add support for XHSC HC32F4A0

Add support for XHSC HC32F4A0

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

XH-YongbingLiu committed:
boards: xhsc: hc32f4a0_evb: Add minimal support for board HC32F4A0_EVB

This adds minimal support for board HC32F4A0_EVB

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

nordicjm commented on west.yml:
```diff
@@ -274,6 +274,11 @@ manifest:
       path: modules/hal/wurthelektronik
       groups:
         - hal
+    - name: hal_xhsc
+      revision: 45f9eec3064663b992da9f1497aaaa8371bcb821
+      path: modules/hal/xhsc
+      groups:
+        - hal
```
I mean the order of commits

decsny commented on dts/bindings/pinctrl/xhsc,hc32-pinctrl.yaml:
```diff
@@ -0,0 +1,55 @@
+
+      drive-strength:
+        type: string
+        default: "low"
```
defaults need to be explained *why* in the description

rruuaanng reviewed (commented):
Some appearance tips.

XH-YongbingLiu committed:
MAINTAINERS: Add maintainers for hal_xhsc.

Add XH-YongbingLiu as maintainers for hal_xhsc.

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>

mbolivar reviewed (commented):
No objections to the new vendor prefix

XH-YongbingLiu committed:
drivers: serial: Remove dma support

The serial DMA feature has been temporarily removed for the
convenience of the PR reviewers. This functionality will be added
back in the next PR.

Signed-off-by: Yongbing Liu <liuyongbing@outlook.com>
